pkg/proxy/proxier.go

```diff
@@ -319,38 +319,43 @@ type Proxier struct {
 	hostIP        net.IP
 }
 
+var (
+	ErrProxyOnLocalhost = fmt.Errorf("cannot proxy on localhost")
```

Comment:
Shouldn't you pre-validate that, then, rather than surfacing it from the depths of the app?  It's a pretty specific error to catch, you might as well check it yourself.